### PR TITLE
Add frontmatter to Triggers API docs

### LINKED ISSUE #1306: Add frontmatter to API docs
### Feature request

Add frontmatter to [docs/triggers-api](https://github.com/tektoncd/triggers/blob/main/docs/triggers-api.md) so it can be included in the [website docs](https://tekton.dev/vault/triggers-main/). The frontmatter would look something like this:

```
<!--
---
title: Triggers API
linkTitle: Triggers API
---
-->
<p>Packages:</p>
...
```

### Use case

The current file can not be included in the website docs because it has no frontmatter. This issue also blocks updating the website theme to a later version, this is causing the build in [this PR](https://github.com/tektoncd/website/pull/339) to fail. The latest versions of Hugo fail if there's a html-only file.

### PR DESCRIPTION
**Pull request by khrm**
# Changes
Fixes #1306

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Comment by dibyom** [comment]
Integration test failures look like they are unrelated - the installation did not come up

/test pull-tekton-triggers-integration-tests

File: hack/update-reference-docs.sh [hunk 1]
```diff
@@ -23,3 +23,4 @@ go run github.com/ahmetb/gen-crd-api-reference-docs \
     -api-dir "github.com/tektoncd/triggers/pkg/apis" \
     -template-dir "./hack/reference-docs-template" \
     -out-file "./docs/triggers-api.md"
+sed -i '1s/^/<!--\n---\ntitle: Triggers API\nlinkTitle: Triggers API\nweight: 1000\n---\n-->\n\n/' ./docs/triggers-api.md
```
Comment (dibyom) [inline hack/update-reference-docs.sh:26]: why do we need this?

Comment (khrm) [reply hack/update-reference-docs.sh:26]: We need this to add frontmatter. There's no option in `github.com/ahmetb/gen-crd-api-reference-docs`. So I added it to this script.

**Pull request merged**